LibreNMS/Alert/Transport/Api.php

```diff
@@ -25,42 +26,57 @@
 namespace LibreNMS\Alert\Transport;
 
 use LibreNMS\Alert\Transport;
+use GuzzleHttp\Client;
 
 class Api extends Transport
 {
     public function deliverAlert($obj, $opts)
     {
         $url = $this->config['api-url'];
+        $options = $this->config['api-options'];
         $method = $this->config['api-method'];
-        return $this->contactAPI($obj, $url, $method);
+        $auth = [$this->config['api-auth-username'], $this->config['api-auth-password']];
+        return $this->contactAPI($obj, $url, $options, $method, $auth);
     }
 
-    private function contactAPI($obj, $api, $method)
+    private function contactAPI($obj, $api, $options, $method, $auth)
     {
         $method = strtolower($method);
-        list($host, $api) = explode("?", $api, 2);
-        foreach ($obj as $k => $v) {
-            $api = str_replace("%" . $k, $method == "get" ? urlencode($v) : $v, $api);
+        $host = explode("?", $api, 2)[0]; //we don't use the parameter part, cause we build it out of options.
+
+        //get each line of key-values and process the variables;
+        foreach (preg_split("/\\r\\n|\\r|\\n/", $options) as $current_line) {
+            list($u_key, $u_val) = explode('=', $current_line, 2);
+            foreach ($obj as $p_key => $p_val) {
+                $u_val = str_replace("{{ $" . $p_key . ' }}', $p_val, $u_val);
```

Comment:
Do we need to escape the values?  I don't think so, they should be escaped by guzzle later...

Reply:
Nope, Guzzle does it so I only str_replace the values here, no escaping.